checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,363 @@
+    if (isDeclaredInTryWithResources(elt)) {
+      type.replaceAnnotation(withoutClose(type.getAnnotationInHierarchy(TOP)));
+    }
+  }
+
+  /**
+   * Returns a {@literal @}MustCall annotation that is like the input, but it does not have "close".
+   * Returns the argument annotation mirror (not a new one) if the argument doesn't have "close" as
+   * one of its elements.
+   *
+   * <p>If the argument is null, returns bottom.
+   *
+   * @param anno a MustCall annotation
+   * @return a MustCall annotation that does not have "close" as one of its values, but is otherwise
+   *     identical to anno
+   */
+  // Package private to permit usage from the visitor in the common assignment check.
+  /* package-private */ AnnotationMirror withoutClose(@Nullable AnnotationMirror anno) {
+    // shortcut for easy paths
+    if (anno == null || AnnotationUtils.areSame(anno, BOTTOM)) {
+      return BOTTOM;
+    } else if (!AnnotationUtils.areSameByName(
+        anno, "org.checkerframework.checker.mustcall.qual.MustCall")) {
+      return anno;
+    }
+    List<String> values =
+        AnnotationUtils.getElementValueArray(anno, mustCallValueElement, String.class);
+    if (!values.contains("close")) {
+      return anno;
+    }
+    return createMustCall(values.stream().filter(s -> !"close".equals(s)).toArray(String[]::new));
+  }
+
+  /**
+   * Returns true iff the given element represents a variable that was declared in a
+   * try-with-resources statement.
+   *
+   * @param elt an element; may be null, in which case this method always returns false
+   * @return true iff the given element represents a variable that was declared in a
+   *     try-with-resources statement
+   */
+  private boolean isDeclaredInTryWithResources(@Nullable Element elt) {
+    return elt != null && elt.getKind() == ElementKind.RESOURCE_VARIABLE;
+  }
+
+  /** Treat non-owning method parameters as @MustCallUnknown when the method is called. */
+  @Override
+  public void methodFromUsePreSubstitution(ExpressionTree tree, AnnotatedExecutableType type) {
+    ExecutableElement declaration;
+    if (tree instanceof MethodInvocationTree) {
+      declaration = TreeUtils.elementFromUse((MethodInvocationTree) tree);
+    } else if (tree instanceof MemberReferenceTree) {
+      declaration = (ExecutableElement) TreeUtils.elementFromTree(tree);
+    } else {
+      throw new BugInCF("unexpected type of method tree: " + tree.getKind());
+    }
+    changeNonOwningParameterTypesToTop(declaration, type);
+    super.methodFromUsePreSubstitution(tree, type);
+  }
+
+  @Override
+  protected void constructorFromUsePreSubstitution(
+      NewClassTree tree, AnnotatedExecutableType type) {
+    ExecutableElement declaration = TreeUtils.elementFromUse(tree);
+    changeNonOwningParameterTypesToTop(declaration, type);
+    super.constructorFromUsePreSubstitution(tree, type);
+  }
+
+  /**
+   * Changes the type of each parameter not annotated as @Owning to top. Also replaces the component
```

Comment:
Receivers cannot be owning, because an object cannot be responsible for calling its must-call methods on itself (then they wouldn't be must-call methods, and the method for which the receiver is "owning" would be must-call instead!).

I've updated the documentation of this method to be explicit on this point.